PWGHF/Tasks/HFMCValidation.cxx

```diff
@@ -79,42 +105,48 @@ struct ValidationGenLevel {
         switch (particlePdgCode) {
           case kCharm:
             cPerCollision++;
+            registry.fill(HIST("hPtVsYCharmQuark"), particle.pt(), particle.y());
             break;
           case kCharmBar:
             cBarPerCollision++;
+            registry.fill(HIST("hPtVsYCharmQuark"), particle.pt(), particle.y());
             break;
           case kBottom:
             bPerCollision++;
+            registry.fill(HIST("hPtVsYBeautyQuark"), particle.pt(), particle.y());
             break;
           case kBottomBar:
             bBarPerCollision++;
+            registry.fill(HIST("hPtVsYBeautyQuark"), particle.pt(), particle.y());
             break;
         }
       }
 
-      sumPxDau = 0;
-      sumPyDau = 0;
-      sumPzDau = 0;
-      momentumCheck = 1;
+      double sumPxDau = 0.;
+      double sumPyDau = 0.;
+      double sumPzDau = 0.;
+      bool momentumCheck = 1;
```

Comment:
momentumCheck is defined as bool but always used as an int, of course it will work but would be better to be consistent